docker/docker_client.go

```diff
@@ -68,14 +87,14 @@ func newDockerClient(ctx *types.SystemContext, ref dockerReference, write bool)
 			tlsc.Certificates = append(tlsc.Certificates, cert)
 		}
 		tlsc.InsecureSkipVerify = ctx.DockerInsecureSkipTLSVerify
-		tr = &http.Transport{
-			TLSClientConfig: tlsc,
-		}
+		tr.TLSClientConfig = tlsc
 	}
 	client := &http.Client{}
-	if tr != nil {
-		client.Transport = tr
+	if tr.TLSClientConfig == nil {
+		var cfg = tlsconfig.ServerDefault
+		tr.TLSClientConfig = &cfg
```

Comment:
Why? In the other path we are happy enough to keep the empty default. Also using `ServerDefault` for a client seems strange (though I realize the client options may be far too restrictive for interoperability).